src/main/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistency.java

```diff
@@ -47,8 +52,17 @@ public Builder auroraConnectionDetails(AuroraConnectionDetails auroraConnectionD
             return this;
         }
 
+        public Builder connectionPerReplicaProvider(BiFunction<String, Properties, AuroraConnectionSupplier> connectionPerReplicaProvider) {
```

Comment:
You can mark `public Builder auroraConnectionDetails(AuroraConnectionDetails auroraConnectionDetails) {` as deprecated and transform AuroraConnectionDetails into connectionPerReplicaProvider here.

Reply:
done